### PR TITLE
Simplify LogVerifier tests

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
This PR is also a preparation for more rigorous testing in #1102 by means of the factored out inclusion verification probes.

### CODE REVIEW

File: merkle/log_verifier_test.go [hunk 1]
```diff
@@ -118,228 +118,184 @@ var (
...
-		return err
-	}
-	if want := root; !bytes.Equal(got, want) {
-		return fmt.Errorf("got root:\n%x\nexpected:\n%x", got, want)
-	}
-	if err := v.VerifyInclusionProof(leafIndex, treeSize, proof, root, leafHash); err != nil {
-		return err
-	}
+// inclVerifProbe is a parameter set for inclusion proof verification.
+type inclVerifProbe struct {
```
Comment (Martin2112) [inline merkle/log_verifier_test.go]: Not sure about the name. What about inclusionProbe? or incProbe? Same for consistency probe.

Comment (pavelkalinnikov) [reply merkle/log_verifier_test.go]: Done.

**Comment by pavelkalinnikov** [comment]
Yeah, good point, will do.

**Comment by pavelkalinnikov** [comment]
PTAL. Did some more refactoring, including `t.Run` wrapping.

**Comment by pavelkalinnikov** [comment]
ping

**Comment by pavelkalinnikov** [comment]
PTAL. This blocks me from using corrupted inclusion verification parameters in #1108 tests.

File: merkle/log_verifier_test.go [hunk 2]
```diff
@@ -17,402 +17,330 @@ package merkle
...
-		return errors.New("incorrectly verified against WrongLeaf")
-	}
+	desc string
+}
 
-	// Wrong root
-	if err := v.VerifyInclusionProof(leafIndex, treeSize, proof, sha256EmptyTreeHash, leafHash); err == nil {
-		return errors.New("incorrectly verified against empty root hash")
-	}
+func corruptInclVerification(leafIndex, treeSize int64, proof [][]byte, root, leafHash []byte) []inclusionProbe {
```
Comment (AlCutter) [inline merkle/log_verifier_test.go]: rename to `corruptInclusionProof`, the `...Verification` is a bit misleading?
(same for `corruptConsistencyProof` below, if you think it's a good idea)

**Comment by AlCutter** [comment]
Just the one suggestion, otherwise LGTM - I'll approve now so as not to hold you up.

File: merkle/log_verifier_test.go (hunk 2 above)
Comment (pavelkalinnikov) [reply merkle/log_verifier_test.go]: Done.

**Comment by pavelkalinnikov** [comment]
Strange. There was no conflicts, but GitHub pushed me to merge.

**Pull request merged**